meter/aggregate_registry.cc

```diff
@@ -0,0 +1,93 @@
+#include "aggregate_registry.h"
+#include "../util/logger.h"
+#include "id_format.h"
+
+using atlas::util::Logger;
+
+namespace atlas {
+namespace meter {
+
+enum class AggrOp { Add, Max };
+
+static AggrOp op_for_statistic(util::StrRef stat) {
+  static std::array<util::StrRef, 5> counters{
+      util::intern_str("count"), util::intern_str("totalAmount"),
+      util::intern_str("totalTime"), util::intern_str("totalOfSquares"),
+      util::intern_str("percentile")};
+
+  auto c = std::find(counters.begin(), counters.end(), stat);
+  return c == counters.end() ? AggrOp::Max : AggrOp::Add;
+}
+
+static AggrOp aggregate_op(const Measurement& measurement) {
+  static util::StrRef statistic = util::intern_str("statistic");
+  const auto& tags = measurement.id->GetTags();
+  auto stat = tags.at(statistic);
+  return op_for_statistic(stat);
+}
+
+static double addnan(double a, double b) {
+  if (std::isnan(a)) {
+    return b;
+  }
+  if (std::isnan(b)) {
+    return a;
+  }
+  return a + b;
+}
+
+static double maxnan(double a, double b) {
+  if (std::isnan(a)) {
+    return b;
+  }
+  if (std::isnan(b)) {
+    return a;
+  }
+  return std::max(a, b);
+}
+
+AggregateRegistry::AggregateRegistry(int64_t update_frequency,
```

Comment:
ConsolidationRegistry might be a better choice. AggregateRegistry could imply something that sends to the inline aggregator service and aggregates across dimensions.